obsidian-tasks-group/obsidian-tasks
# refactor: Create `EditableTask` class

# Types of changes

Internal changes:


## Description

- Convert `type EditableTask` to `class EditableTask`
- Move related code from EditTask modal to Editable Task class

## Motivation and Context

- Remove logic from Edit Task modal
- Encapsulate behaviour in Editable Task

## How has this been tested?

- Smoke test in demo vault
  - create task with just a description from an empty line, close modal, verify that the task has been written correctly
  - modify every field, close modal, verify that the task has been written correctly
  - remove all field data except description, close modal, verify that the task has been written correctly

## Checklist

  - hence the smoke test

## Terms
